### ISSUE TITLE
No `POST` processing and subsequent crash

### ISSUE THREAD

**Issue by krizhanovsky** [tempesta-tech/tempesta#2155]
Tempesta master as of 78236a251ff34d73fece8c77afdf3a0797fa1429, the linux kernel 5.10.35-tfw as of 09f6e2a043d099627d0617dbe34c2d1521540927 . `tempesta-1` vm on our website with config
```
listen 443 proto=h2,https;

cache 1;
cache_size 1073741824;
cache_fulfill * *;
cache_methods GET HEAD;
cache_ttl 3600;
cache_purge;
# Allow purging from both the containers (production and staging upstreams),
# localhost (VM) and the host. Since in the current configuration the same
# IP is used for both the container and the host system, only one (production)
# address is configured.
cache_purge_acl 192.168.122.1 127.0.0.1;

access_log on;

frang_limits {
        request_rate 200;
        http_method_override_allowed true;
        http_methods post put get;
}

# Allow only following characters in URI: %+,/a-zA-Z0-9&?:-.[]_=
# These are tested with the WordPress admin panel.
http_uri_brange 0x25 0x2b 0x2c 0x2f 0x41-0x5a 0x61-0x7a 0x30-0x39 0x26 0x3f 0x3a 0x2d 0x2e 0x5b 0x5d 0x5f 0x3d;

health_stat 3* 4* 5*;
health_stat_server 3* 4* 5*;

block_action attack reply;
block_action error reply;

# Make WordPress to work over TLS.
# See https://tempesta-tech.com/knowledge-base/WordPress-tips-and-tricks/
req_hdr_add X-Forwarded-Proto "https";

resp_hdr_set Strict-Transport-Security "max-age=31536000; includeSubDomains";

# Remove the proxy header to mitigate the httpoxy vulnerability
# See https://httpoxy.org/
req_hdr_set Proxy;

tls_certificate /etc/tempesta/fullchain.pem;
tls_certificate_key /etc/tempesta/privkey.pem;

srv_group production {
        server 192.168.122.1:8001;
}

srv_group staging {
        server 192.168.122.1:8000;
}

vhost tempesta-tech.com {
        proxy_pass production;
}

vhost staging.tempesta-tech.com {
        proxy_pass staging;
}

http_chain staging_chain {
	# Disable caching for staging to avoid sophisticated problems with
        # purging WP themes and other utilities, which might affect all resources
        # or, worse, resources hard to detect. Meantime performance for staging
        # isn't crucial.
        uri == "*" -> cache_disable;
	-> staging.tempesta-tech.com;
}

http_chain {
	# Redirect old URLs from the old static website
	uri == "/index"		-> 301 = /;
	uri == "/development-services" -> 301 = /network-security-performance-analysis;

        # Disable PHP dynamic logic for caching
        # See https://www.varnish-software.com/developers/tutorials/configuring-varnish-wordpress/
        uri == "/wp-admin*" -> cache_disable;
        uri == "/wp-comments-post.php*" -> cache_disable;

        # RSS feed /comments/feed/ is cached as other resource for 1 hour,
        # defined by the global cache_ttl policy.

	# Proably outdated redirects
	uri == "/index.html"	-> 301 = /;
	uri == "/services"	-> 301 = /development-services;
	uri == "/services.html"	-> 301 = /development-services;
	uri == "/c++-services"	-> 301 = /development-services;
	uri == "/company.html"	-> 301 = /company;
	uri == "/blog/fast-programming-languages-c-c++-rust-assembly" -> 301 = /blog/fast-programming-languages-c-cpp-rust-assembly;

	mark == 1 -> staging_chain;

	-> tempesta-tech.com;
}
```

Login to the admin console and go to plugins, try to update all the plugins (choose `Update` bulk operation on `https://tempesta-tech.com/wp-admin/plugins.php?plugin_status=upgrade` and press `Apply`). A browswer receives response 200 on `POST /wp-admin/admin-ajax.php`, but nothing happens (the update doesn't take place). In several tries I got crash
```
[ 2064.260384] [tempesta fw] 46.6.46.45 "tempesta-tech.com" "POST /wp-admin/admin-ajax.php HTTP/2.0" 200 180 "https://tempesta-tech.com/wp-admin/plugins.php?plugin_status=upgrade" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:127.0) Gecko/20100101 Firefox/127.0"
[ 2067.740662] [tempesta fw] 95.108.213.122 "tempesta-tech.com" "GET /sitemap_index.xml HTTP/1.1" 200 366 "-" "Mozilla/5.0 (compatible; YandexBot/3.0; +http://yandex.com/bots)"
[ 2129.067144] BUG: unable to handle page fault for address: 0000000000077f80
[ 2129.068425] #PF: supervisor read access in kernel mode
[ 2129.069357] #PF: error_code(0x0000) - not-present page
[ 2129.070286] PGD 0 P4D 0 
[ 2129.070854] Oops: 0000 [#1] SMP
[ 2129.071492] CPU: 9 PID: 0 Comm: swapper/9 Tainted: G           O      5.10.35-tfw-secperf+ #4
[ 2129.072939] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS 1.15.0-1 04/01/2014
[ 2129.074328] RIP: 0010:tfw_tls_on_send_alert+0x2d/0xb0 [tempesta_fw]
[ 2129.075425] Code: f5 53 48 89 fb 48 81 c7 b0 04 00 00 e8 bc de f2 ff 84 c0 74 56 48 81 fb d0 f7 ff ff 74 4d 48 8b 83 98 09 00 00 48 85 c0 74 46 <48> 8b 90 a0 03 00 00 48 85 d2 74 3a 48 8b 75 00 48 8b 4a 08 48 8b
[ 2129.077108] RSP: 0018:ffffb748002a0f00 EFLAGS: 00010202
[ 2129.077108] RAX: 0000000000077be0 RBX: ffff9b18f3d75040 RCX: 0000000000000001
[ 2129.077108] RDX: 0000000000000002 RSI: ffffb748002a0f38 RDI: ffff9b18f3d754f0
[ 2129.077108] RBP: ffffb748002a0f38 R08: ffff9b18f3d75040 R09: 0000000000000020
[ 2129.077108] R10: 0000000000000008 R11: ffffffffa1c060c0 R12: 000000000002c640
[ 2129.077108] R13: 000000000000152c R14: ffff9b18f0d04c00 R15: 0000000000000576
[ 2129.077108] FS:  0000000000000000(0000) GS:ffff9b1bafa40000(0000) knlGS:0000000000000000
[ 2129.077108] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 2129.077108] CR2: 0000000000077f80 CR3: 00000001e1e0a005 CR4: 0000000000770ee0
[ 2129.077108] PKRU: 55555554
[ 2129.077108] Call Trace:
[ 2129.077108]  <IRQ>
[ 2129.077108]  ss_tx_action+0x40f/0x740 [tempesta_fw]
[ 2129.077108]  net_tx_action+0x7e/0x1f0
[ 2129.077108]  __do_softirq+0xc0/0x26e
[ 2129.077108]  asm_call_irq_on_stack+0xf/0x20
[ 2129.077108]  </IRQ>
[ 2129.077108]  do_softirq_own_stack+0x32/0x40
[ 2129.077108]  irq_exit_rcu+0x83/0xb0
[ 2129.077108]  sysvec_apic_timer_interrupt+0x2e/0x80
[ 2129.077108]  asm_sysvec_apic_timer_interrupt+0x12/0x20
[ 2129.077108] RIP: 0010:default_idle+0xe/0x10
[ 2129.077108] Code: fb 65 48 8b 04 25 00 6d 01 00 f0 80 60 02 df c3 0f ae f0 0f ae 38 0f ae f0 eb b6 90 e9 07 00 00 00 0f 00 2d 06 c8 47 00 fb f4 <c3> cc 65 8b 15 b9 89 c8 5e 89 d2 48 8b 05 20 26 67 01 48 03 04 d5
[ 2129.077108] RSP: 0018:ffffb748000b3ef0 EFLAGS: 00000202
[ 2129.077108] RAX: ffff9b1bafa6ab00 RBX: ffff9b1840235700 RCX: ffff9b1bafa6ab00
[ 2129.077108] RDX: 0000000000025136 RSI: 0000000000000083 RDI: 0000000000000009
[ 2129.077108] RBP: 0000000000000000 R08: ffff9b1bafa5dc40 R09: 0000000000000029
[ 2129.077108] R10: 0000000000000001 R11: 0000000000000000 R12: 0000000000000000
[ 2129.077108] R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000000000
[ 2129.077108]  default_idle_call+0x2d/0xa0
[ 2129.077108]  cpuidle_idle_call+0x133/0x190
[ 2129.077108]  ? tsc_verify_tsc_adjust+0x2d/0xc0
[ 2129.077108]  do_idle+0x68/0xb0
[ 2129.077108]  cpu_startup_entry+0x14/0x20
[ 2129.077108]  secondary_startup_64_no_verify+0xb0/0xbb
[ 2129.077108] Modules linked in: tcp_diag inet_diag tempesta_fw(O) tempesta_db(O) tempesta_tls(O) tempesta_lib(O) serio_raw sg virtio_balloon virtio_console button loop sch_fq_codel msr drm efi_pstore configfs ip_tables x_tables ext4 crc16 mbcache jbd2 efivars linear md_mod sr_mod cdrom virtio_net virtio_rng net_failover rng_core virtio_blk failover ahci libahci libata crct10dif_pclmul scsi_mod i2c_i801 lpc_ich psmouse i2c_smbus mfd_core virtio_pci virtio_ring virtio [last unloaded: tempesta_lib]
[ 2129.077108] CR2: 0000000000077f80
[ 2129.077108] ---[ end trace 133bbd570d38e9b1 ]---
```

**Comment by krizhanovsky** [comment]
Caught the same bug on my 4cpu VM and just 5 curl requests:
```
curl -Lvk --http2-prior-knowledge https://tempesta-tech.com
curl -Lvk --http2-prior-knowledge --output /dev/null https://tempesta-tech.com 
curl -Lvk --http2-prior-knowledge --tlsv1.2 --tls-max 1.2 https://tempesta-tech.com
curl -Lvk https://tempesta-tech.com
curl -Lvk --http1.1 https://tempesta-tech.com
```

```
[ 6191.582394] [tempesta fw] 192.168.100.1 "default" "GET / HTTP/2.0" 200 0 "-" "curl/7.81.0"
[ 6202.671689] [tempesta fw] 192.168.100.1 "default" "GET / HTTP/2.0" 200 0 "-" "curl/7.81.0"
[ 6224.918442] [tempesta fw] 192.168.100.1 "default" "GET / HTTP/2.0" 200 0 "-" "curl/7.81.0"
[ 6277.638254] [tempesta fw] 192.168.100.1 "default" "GET / HTTP/1.1" 200 0 "-" "curl/7.81.0"
[ 6277.664785] general protection fault, probably for non-canonical address 0xb404cf95bbaa5b62: 0000 [#1] SMP
[ 6277.665478] CPU: 3 PID: 0 Comm: swapper/3 Kdump: loaded Tainted: G           O      5.10.35-tfw-secperf+ #5
[ 6277.666134] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.15.0-1 04/01/2014
[ 6277.666707] RIP: 0010:tfw_tls_on_send_alert+0x2d/0xb0 [tempesta_fw]
[ 6277.667111] Code: f5 53 48 89 fb 48 81 c7 b0 04 00 00 e8 bc be f1 ff 84 c0 74 56 48 81 fb d0 f7 ff ff 74 4d 48 8b 83 98 09 00 00 48 85 c0 74 46 <48> 8b 90 a0 03 00 00 48 85 d2 74 3a 48 8b 75 00 48 8b 4a 08 48 8b
[ 6277.668300] RSP: 0018:ffffbc2740118f00 EFLAGS: 00010282
[ 6277.668642] RAX: b404cf95bbaa5b62 RBX: ffff9c12ca2fda48 RCX: 0000000000000008
[ 6277.669106] RDX: 0000000000000100 RSI: ffffbc2740118f38 RDI: ffff9c12ca2fdef8
[ 6277.669586] RBP: ffffbc2740118f38 R08: ffff9c12ca2fda48 R09: 0000000000000020
[ 6277.670056] R10: 0000000000000008 R11: ffffffff96c060c0 R12: 000000000002c780
[ 6277.670547] R13: 000000000000150c R14: ffff9c12c9ecaf80 R15: 00000000000005a8
[ 6277.671048] FS:  0000000000000000(0000) GS:ffff9c13b7d80000(0000) knlGS:0000000000000000
[ 6277.671563] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 6277.671938] CR2: 00007fb77fa40930 CR3: 000000014521d000 CR4: 0000000000750ee0
[ 6277.672395] PKRU: 55555554
[ 6277.672586] Call Trace:
[ 6277.672791]  <IRQ>
[ 6277.672970]  ss_tx_action+0x40f/0x740 [tempesta_fw]
[ 6277.673340]  net_tx_action+0x7e/0x1f0
[ 6277.673630]  __do_softirq+0xc0/0x26e
[ 6277.673875]  asm_call_irq_on_stack+0xf/0x20
[ 6277.674174]  </IRQ>
[ 6277.674338]  do_softirq_own_stack+0x32/0x40
[ 6277.674675]  irq_exit_rcu+0x83/0xb0
[ 6277.674923]  sysvec_call_function_single+0x2e/0x80
[ 6277.675747]  asm_sysvec_call_function_single+0x12/0x20
[ 6277.676424] RIP: 0010:default_idle+0xe/0x10
[ 6277.677008] Code: fb 65 48 8b 04 25 00 6d 01 00 f0 80 60 02 df c3 0f ae f0 0f ae 38 0f ae f0 eb b6 90 e9 07 00 00 00 0f 00 2d b6 b6 47 00 fb f4 <c3> cc 65 8b 15 69 78 c8 69 89 d2 48 8b 05 d0 14 67 01 48 03 04 d5
[ 6277.678872] RSP: 0018:ffffbc274008bef0 EFLAGS: 00000212
[ 6277.679527] RAX: ffff9c13b7daac00 RBX: ffff9c128018ba00 RCX: ffff9c13b7daac00
[ 6277.680294] RDX: 000000000003d33e RSI: 0000000000000087 RDI: 0000000000000003
[ 6277.681443] RBP: 0000000000000000 R08: ffff9c13b7d9dc40 R09: 000000000000011e
[ 6277.682209] R10: 0000000000000001 R11: 0000000000000001 R12: 0000000000000000
[ 6277.682968] R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000000000
[ 6277.683727]  default_idle_call+0x2d/0xa0
[ 6277.684302]  cpuidle_idle_call+0x133/0x190
[ 6277.684871]  ? tsc_verify_tsc_adjust+0x2d/0xc0
[ 6277.685478]  do_idle+0x68/0xb0
[ 6277.685978]  cpu_startup_entry+0x14/0x20
[ 6277.686558]  secondary_startup_64_no_verify+0xb0/0xbb
[ 6277.687169] Modules linked in: tcp_diag inet_diag tempesta_fw(O) tempesta_db(O) tempesta_tls(O) tempesta_lib(O) bridge stp ll
```

**Comment by krizhanovsky** [comment]
Also reproduced this crash just with running our TLS tests suite:
```
root@ubuntu:~/tempesta-test# ./run_tests.py -n tls
Root privileges are required: need access to module loading on localhost.

----------------------------------------------------------------------
Running functional tests...
----------------------------------------------------------------------

test (tls.test_tls_tickets.TlsVhostConfusionDfltVhost) ... ok
test (tls.test_tls_tickets.TlsVhostConfusionDfltCertsWithUnknown) ... ok
test (tls.test_tls_tickets.TlsVhostConfusionDfltCerts) ... 
```
```
[44305.726865] [tempesta fw] Tempesta FW is ready
[44305.977209] device ens3 entered promiscuous mode
[44306.677755] BUG: unable to handle page fault for address: 00000000021c03a0
[44306.686584] #PF: supervisor read access in kernel mode
[44306.686584] #PF: error_code(0x0000) - not-present page
[44306.686584] PGD 0 P4D 0
[44306.686584] Oops: 0000 [#1] SMP
[44306.686584] CPU: 1 PID: 5533 Comm: python3 Kdump: loaded Tainted: G           O      5.10.35-tfw-secperf+ #5
[44306.686584] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.15.0-1 04/01/2014
[44306.686584] RIP: 0010:tfw_tls_on_send_alert+0x2d/0xb0 [tempesta_fw]
[44306.686584] Code: f5 53 48 89 fb 48 81 c7 b0 04 00 00 e8 cc de f2 ff 84 c0 74 56 48 81 fb d0 f7 ff ff 74 4d 48 8b 83 98 09 00 00 48 85 c0 74 46 <48> 8b 90 a0 03 00 00 48 85 d2 74 3a 48 8b 75 00 48 8b 4a 08 48 8b
[44306.686584] RSP: 0018:ffffa6a2800c0f00 EFLAGS: 00010206
[44306.686584] RAX: 00000000021c0000 RBX: ffff8b4eafce4180 RCX: 0000000000000001
[44306.686584] RDX: 0000000000000002 RSI: ffffa6a2800c0f38 RDI: ffff8b4eafce4630
[44306.686584] RBP: ffffa6a2800c0f38 R08: ffff8b4eafce4180 R09: 0000000000000020
[44306.686584] R10: 0000000000000008 R11: ffffffff946060c0 R12: 000000000002c780
[44306.686584] R13: 000000000000150c R14: ffff8b4ea6da6880 R15: 0000000000008000
[44306.686584] FS:  00007fe9053c4640(0000) GS:ffff8b4fb7c80000(0000) knlGS:0000000000000000
[44306.686584] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[44306.686584] CR2: 00000000021c03a0 CR3: 0000000126665000 CR4: 0000000000750ee0
[44306.686584] PKRU: 55555554
[44306.686584] Call Trace:
[44306.686584]  <IRQ>
[44306.686584]  ss_tx_action+0x40f/0x740 [tempesta_fw]
[44306.686584]  net_tx_action+0x7e/0x1f0
[44306.686584]  __do_softirq+0xc0/0x26e
[44306.686584]  asm_call_irq_on_stack+0xf/0x20
[44306.686584]  </IRQ>
[44306.686584]  do_softirq_own_stack+0x32/0x40
[44306.686584]  do_softirq+0x42/0x50
[44306.686584]  __local_bh_enable_ip+0x46/0x50
[44306.686584]  ip_finish_output2+0x18a/0x510
[44306.686584]  ? __ip_finish_output.part.0+0xb9/0x1c0
[44306.686584]  __ip_queue_xmit+0x1a8/0x420
[44306.686584]  __tcp_transmit_skb+0x4be/0x8f0
[44306.686584]  tcp_write_xmit+0x3f6/0x9d0
[44306.686584]  __tcp_push_pending_frames+0x2d/0xb0
[44306.686584]  tcp_sendmsg_locked+0x9fa/0xb10
[44306.686584]  tcp_sendmsg+0x23/0x40
[44306.686584]  sock_sendmsg+0x2e/0x40
[44306.686584]  __sys_sendto+0xd4/0x120
[44306.686584]  ? do_futex+0x79/0x130
[44306.686584]  ? __x64_sys_futex+0x6e/0x1a0
[44306.686584]  ? ktime_get_real_ts64+0x44/0xe0
[44306.686584]  ? _copy_to_user+0x1c/0x30
[44306.686584]  ? put_timespec64+0x27/0x40
[44306.686584]  __x64_sys_sendto+0x1b/0x20
[44306.686584]  do_syscall_64+0x2d/0x70
[44306.686584]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[44306.686584] RIP: 0033:0x7fe90a66f8fe
[44306.686584] Code: 4c 24 1c e8 14 91 f6 ff 44 8b 54 24 1c 8b 3c 24 45 31 c9 41 89 c4 48 8b 54 24 10 48 8b 74 24 08 45 31 c0 b8 2c 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 32 44 89 e7 48 89 04 24 e8 4e 91 f6 ff 48 8b
```

**Comment by krizhanovsky** [comment]
The crash is fixed in #2158 , but plugins update doesn't work.

**Comment by EvgeniiMekhanik** [comment]
I am not sure that problem with plugins update is in Tempesta FW. First of all I check that POST request works fine and I can add comments without any problems. Then I try to reset last commits (reworking making frames and stream prioritization implementation) and I can't update plugins also. After I press the `apply` button, mark from plugins removed.
![Снимок экрана от 2024-07-02 19-48-10](https://github.com/tempesta-tech/tempesta/assets/122307047/2141c65b-8264-42a4-808d-26dbfb9f0849)

**Comment by krizhanovsky** [comment]
It seems for me also that the problem is with WP or particular plugins. I tried to update the plugins in 3 different ways on the production container. The first one updated several plugins in a row:

![Screenshot_2024-07-05_15-57-16](https://github.com/tempesta-tech/tempesta/assets/884694/601525bb-6aa4-41b0-9d99-8ca34f80ad08)

Next, I tried to reproduce the bulk action as above and it failed on the first of the rest plugins since it's a commercial one:

![Screenshot_2024-07-05_16-03-22](https://github.com/tempesta-tech/tempesta/assets/884694/41e6aec9-af73-4e9b-ad60-2b6a934c2ac0)

In the screenshot above the 3rd plugin update hang (but the action is started, i.e. POST was processed on WP side). And after that I could successfully update the plugin individually.

![Screenshot_2024-07-05_16-03-22](https://github.com/tempesta-tech/tempesta/assets/884694/6b3afca4-86b9-4e1c-b76e-ff0983f0a25c)

I believe the issue can be closed now.
